### PR TITLE
feat: add Clang and Black formatting tools

### PR DESCRIPTION
**Pull request by kyle-cochran**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,7 +17,7 @@ jupyter_notebook_port := 9003
 jupyter_python_version := 3.8
 jupyter_project_name_python_shared_object := OpenSpaceToolkitMathematicsPy.cpython-38-x86_64-linux-gnu
 
-################################################################################################################################################################
+clang_format_sources_path ?= $(shell find ~+ src/ include/ test/ bindings/python/src/ -name '*.cpp' -o -name '*.cxx' -o -name '*.hpp')
```
Comment (Derollez) [inline Makefile:20]:
```suggestion
clang_format_sources_path ?= $(shell find ~+ src/ include/ test/ bindings/python/src/ -name '*.cpp' -o -name '*.cxx' -o -name '*.hpp' -o -name '*.tpp')
```

Take into account template files as well

File: Makefile [hunk 2]
```diff
@@ -272,6 +262,37 @@ debug-development: build-development-image ## Debug development environment
...
+format-python: build-development-image  ## Runs the black format tool against python code
+
+	docker run \
+		--rm \
+		--privileged \
+		--volume="$(CURDIR):/app:delegated" \
+		--workdir=/app \
+		$(docker_development_image_repository):$(docker_image_version) \
+		/bin/bash -c "python3.11 -m black --line-length=90 bindings/python/"
+
```
Comment (Derollez) [inline Makefile:294]: Use the `.PHONY` per target, instead of having a full list at the bottom to align with astro

**Review by Derollez — APPROVED** [review]
2 tiny modifications and let's merge

**Review by vishwaLoft — APPROVED** [review]
LGTM

**Pull request merged**